tests/cupy_tests/manipulation_tests/test_kind.py

```diff
@@ -32,7 +32,10 @@ def test_asarray_chkfinite_non_finite_vals(self, dtype, order):
     def test_asfarray(self, dtype):
         a = cupy.asarray([1, 2, 3])
         a_gpu = cupy.asfarray(a, dtype)
-        a_cpu = numpy.asfarray(a.get(), dtype)
+        # Original: a_cpu = numpy.asfarray(a.get(), dtype)
+        if not numpy.issubdtype(dtype, numpy.inexact):
+            dtype = numpy.float64
+        a_cpu = numpy.asarray(a.get(), dtype)
         assert a_cpu.dtype == a_gpu.dtype
```

Comment:
Am curious about the context for these changes. Was there a particular issue that cropped up here?

Reply:
`asfarray` was removed because it had weird semantics (by default casting float32 arrays to float64, which is definitely not what you expect.  Basically, to get what you want, you always had to use `asfarray(arr, dtype=arr.dtype)`, I suspect).

So this is good until cupy just deletes `asfarray` as well (at which point the test goes away).